packages/backend/src/controllers/moodle/index.ts

```diff
@@ -14,74 +16,88 @@ import { getRefreshRate } from '../refreshRate/refreshRate';
+export async function fetchAndNotify(): Promise<boolean> {
+  try {
+    const moodleUrl = getBaseUrl();
+    const courseBlacklist: number[] = await getCourseBlacklist();
 
-        const lastFetch = await getLastFetch();
-        if (!lastFetch) throw new Error('Unable to get timestamp of last fetch');
+    const lastFetch = await MoodleSettings.getLastFetch();
 
-        // get all course IDs and map them to corresponding course names
-        const courseDetails = await fetchEnrolledCourses(moodleUrl);
-        const courseMap: Map<number, string> = new Map();
-        courseDetails.forEach(course => courseMap.set(course.id, course.shortname));
+    // get all course IDs and map them to corresponding course names
+    const courseDetails = await fetchEnrolledCourses(moodleUrl);
+    const courseMap: Map<number, string> = new Map();
+    courseDetails.forEach(course => courseMap.set(course.id, course.shortname));
 
-        // fetch and handle course contents
-        for (const courseId of courseMap.keys()) {
-            if (courseBlacklist.includes(courseId)) continue;
-            const content = await fetchCourseContents(moodleUrl, courseId);
-            handleContents(content, courseMap.get(courseId), lastFetch);
-        }
+    // fetch and handle course contents
+    for (const courseId of courseMap.keys()) {
+      if (courseBlacklist.includes(courseId)) continue;
+      const content = await fetchCourseContents(moodleUrl, courseId);
+      handleContents(content, courseMap.get(courseId), lastFetch);
+    }
 
-        const courseList = await fetchAssignments(moodleUrl).then(courses =>
-            courses.filter(course => !courseBlacklist.includes(course.id)));
+    const courseList = await fetchAssignments(moodleUrl).then(courses =>
+      courses.filter(course => !courseBlacklist.includes(course.id)));
 
-        handleAssignments(courseList, lastFetch);
+    handleAssignments(courseList, lastFetch);
 
-    } catch(error) {
-        loggerFile.error('Moodle API request failed', error);
-    }
+    return true;
+  } catch(error) {
+    loggerFile.error('Moodle API request failed', error);
+    return false;
+  } finally {
+    await MoodleSettings.findOneAndUpdate({}, { $set: { lastFetch: Math.floor(Date.now() / 1000) }});
+  }
 }
 
 /**
  * Calls the fetchAndNotify function in the current refresh rate interval.
  *
  * @export
  */
-export async function continuousFetchAndNotify(): Promise<void> {
-    fetchAndNotify();
-    // Call function again after database interval
-    const interval = await getRefreshRate();
-    setTimeout(continuousFetchAndNotify, interval);
+export async function continuousFetchAndNotify(): Promise<boolean> {
+  const wasSuccessful: boolean = await fetchAndNotify();
+
+  // Call function again after database interval
+  const interval = await MoodleSettings.getRefreshRate();
+  setTimeout(continuousFetchAndNotify, interval);
+
+  return wasSuccessful;
+}
```

Comment:
```suggestion
export async function continuousFetchAndNotify(): Promise<void> {
  fetchAndNotify();

  // Call function again after database interval
  const interval = await MoodleSettings.getRefreshRate();
  setTimeout(continuousFetchAndNotify, interval);
}
```

Ich würde hier eher nicht den Status zurückgeben:
1. Er wird nicht gebraucht und nirgends verwendet.
2. Das `await` verfälscht das Fetch-Intervall um die Dauer des Fetches. Unter anderem macht das auch spätere Berechnungen von `nextFetch` deutlich schwieriger.

Reply:
Stimmt, dieser Code kann wieder weg - Vermutlich ein altes Überbleibsel für die API. Kann deine Änderungen leider nicht direkt commiten, weil die Tests dann noch wieder angepasst werden müssen. :grin:

Reply:
Aber sehr cool, dass du es gefunden hast :+1: @antonplagemann